gammapy/modeling/tests/test_serialize_yaml.py

```diff
@@ -138,9 +138,9 @@ def test_datasets_to_io(tmp_path):
         dataset1.model.skymodels[1].parameters["lon_0"].value, 0.9, atol=0.1
     )
 
-    datasets.to_yaml(str(tmp_path / "written_"))
+    datasets.to_yaml(tmp_path, prefix="written")
     datasets_read = Datasets.from_yaml(
-        tmp_path / "written_datasets.yaml", tmp_path / "written_models.yaml"
+        str(tmp_path / "written_datasets.yaml"), str(tmp_path / "written_models.yaml")
```

Comment:
Same here: `Datasets.from_yaml` should work with Path objects. So remove the calls to `str` here.